src/wrappers/loadable.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react'
+import { Router, Route, useParams } from 'react-router-dom'
+import { bindActionCreators } from 'redux'
+import { connect, useSelector } from 'react-redux'
+import { restoreDepositState, restoreRedemptionState } from '../actions';
+import { useWeb3React } from '@web3-react/core'
+
+export const RESTORER = {
+  DEPOSIT: 'deposit',
+  REDEMPTION: 'redemption'
+}
+
+function LoadableBase({ children, account, restoreDepositState, restoreRedemptionState, restorer }) {
+    // Wait for web3 connected
+    const { active: web3Active } = useWeb3React()
+    const { address } = useParams()
+    const depositStateRestored = useSelector((state) => state[restorer].stateRestored)
+    
+    useEffect(() => {
+        if(web3Active) {
+            if (address && ! depositStateRestored) {
+                if (restorer == RESTORER.DEPOSIT) {
+                    restoreDepositState(address)
+                } else if (restorer == RESTORER.REDEMPTION) {
+                    restoreRedemptionState(address)
+                } else {
+                    throw "Unknown restorer."
+                }
+                // return <div>Loading...</div>
+            } else {
+                // FIXME How do we not render these if we're getting ready to transition to
+                // FIXME a new page?
+                // return children
+            }
+        }
+    }, [web3Active])
```

Comment:
Shouldn't this include something around the state we're using for `depositStateRestored`? Also although we don't expect these to change, `restoreDepositState` and `restoreRedemptionState` ought to also have entries here, no?

Reply:
> If you use this optimization, make sure the array includes all values from the component scope (such as props and state) that change over time and that are used by the effect. Otherwise, your code will reference stale values from previous renders. Learn more about how to deal with functions and what to do when the array values change too often.

Yep, you're completely right. Good catch.